### PR TITLE
Add support for deduplication testing

### PR DESCRIPTION
**Pull request by klwilson227**
Add some basic support for testing deduplication feature. 
1) Ability to set the producer name in the test client. 
2) Ability to set the sendTimeout in the test client. 
3) Ability to set the separator used between producername and threadnumber. As well as the topic and topic number. Defaults to '-'
4) Ability to customize format each payload record if enabled. 
    - Ability to  customize/replace the formatting class which is used.
    - DefaultFormatter provided:
    - MetaData replacements: %p replaced with producer Name, %t nanoseconds when replacement occured, %i external messageId 
    - Other replacements follow java sprintf formatting with %d %s %f %l with randomly generated strings or numbers.

### CODE REVIEW

File: pulsar-testclient/pom.xml [hunk 1]
```diff
@@ -94,6 +94,13 @@
 			<groupId>org.hdrhistogram</groupId>
 			<artifactId>HdrHistogram</artifactId>
 		</dependency>
+		<dependency>
+			<groupId>org.json</groupId>
+			<artifactId>json</artifactId>
+			<version>20201115</version>
+			<scope>test</scope>
+		</dependency>
```
Comment (merlimat) [inline pulsar-testclient/pom.xml:102]: The license of this library is `The Software shall be used for Good, not Evil.` and it's not really compatible with Apache license. We generally use Jackson and it's already in class path.

**Review by merlimat — COMMENTED** [review]
If the goal is testing the deduping mechanism, we should probably also add something in the performance consumer to validate that no messages with duplicated sequences are received.

File: pulsar-testclient/pom.xml (hunk 1 above)
Comment (klwilson227) [reply pulsar-testclient/pom.xml:102]: Thanks for the feedback, I will take a look at updating.

Comment (klwilson227) [reply pulsar-testclient/pom.xml:101]: Move completed to jackson.

**Comment by klwilson227** [comment]
Merlimat, RE the consumer side, I'm sure the consumer side would also benefit as well for a possible demo/software acceptance level validation. However, in my particular use case the validation and scanning for duplicates and lost records was easier implemented in the Data Sink which received the data  hence I did not invest extra to build a data validation layer within pulsar test client, only the ability to publish data using both the dedup enabled and dedup disabled.

Since, I did not have the requirement, it was not coded. And I did not want to assume that this was a requirement, as a poor implementation may show both strengths and weaknesses in the platform.  I can certainly take it in that direction. But would think that may be a separate PR.

**Review by merlimat — APPROVED** [review]

**Review by aahmed-se — APPROVED** [review]

**Comment by lhotari** [comment]
This PR introduced a flaky test, reported as #10378. I have created a PR to fix the flakiness in #10379 . Please review.

**Pull request merged**